packages/hardhat-core/test/internal/hardhat-network/helpers/useProvider.ts

```diff
@@ -115,6 +119,20 @@ export function useProvider({
         this.server.getProvider()
       );
     }
+
+    if (rethnetBinary !== undefined) {
+      const { childProcess, processExit, httpProvider } =
```

Comment:
I would rename `processExit` to `isReady`, and document that it's a promise that resolves if the process doesn't fail in N milliseconds.

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/4095/commits/dda3c487a083ea2a719b430a7c09be9596ded35f